### PR TITLE
feat: allow downloading of all pinned browsers

### PR DESCRIPTION
**Pull request by Lightning00Blade**
One thing that is under works is how to handle the configuration? Should we accept another parameter? 

Note: this does not contain breaking changes.

### CODE REVIEW

**Comment by OrKoN** [comment]
> One thing that is under works is how to handle the configuration? Should we accept another parameter?

which configuration?

> Note: this does not contain braking changes.

s/braking/breaking

**Comment by Lightning00Blade** [comment]
So currently it download every browser that is pinned, but from Puppeteer CLI, we would want the configuration file and default to manage which browser get  downloaded.

**Comment by OrKoN** [comment]
@Lightning00Blade, yes, so pinned browsers are only set for the Puppeteer CLI, so we would need to update the list of pinned browsers in packages/puppeteer/src/node/cli.ts based on the current configuration (probably, after https://github.com/puppeteer/puppeteer/pull/12795 lands).

File: packages/browsers/src/CLI.ts [hunk 1]
```diff
@@ -152,9 +153,11 @@ export class CLI {
...
     const latestOrPinned = this.#pinnedBrowsers ? 'pinned' : 'latest';
+    // If there are pinned browser allow the positional arg to be optional
+    const browserArgType = this.#pinnedBrowsers ? '[browser]' : '<browser>';
     return yargs
       .command(
-        'install <browser>',
+        `install ${browserArgType}`,
         'Download and install the specified browser. If successful, the command outputs the actual browser buildId that was installed and the absolute path to the browser executable (format: <browser>@<buildID> <path>).',
         yargs => {
           this.#defineBrowserParameter(yargs);
```
Comment (OrKoN) [inline packages/browsers/src/CLI.ts:173]: let's add an example for installing all pinned browsers (if pinned browsers are known)

Comment (OrKoN) [reply packages/browsers/src/CLI.ts:173]: By example, I mean using `yargs.example()` that happens below these lines

File: packages/browsers/src/CLI.ts [hunk 2]
```diff
@@ -261,50 +264,21 @@ export class CLI {
...
-          if (args.browser.buildId === 'pinned') {
-            const pinnedVersion = this.#pinnedBrowsers?.[args.browser.name];
-            if (!pinnedVersion) {
-              throw new Error(
-                `No pinned version found for ${args.browser.name}`
-              );
-            }
-            args.browser.buildId = pinnedVersion;
+          if (this.#pinnedBrowsers && !args.browser) {
+            await Promise.all(
```
Comment (OrKoN) [inline packages/browsers/src/CLI.ts]: should we use Promise.allSettled here and throw if there is any error?

Comment (Lightning00Blade) [reply packages/browsers/src/CLI.ts]: I think it make sense, one problem with the updated implementation is surfacing the error messages from the install function.

Comment (OrKoN) [reply packages/browsers/src/CLI.ts]: I think re-throwing the first one would result in the same behavior w.r.t to errors

**Review by OrKoN — APPROVED** [review]

**Pull request merged**